drone-webdriver/src/main/java/org/jboss/arquillian/drone/webdriver/factory/remote/reusable/ReusableRemoteWebDriver.java

```diff
@@ -80,8 +81,20 @@ public static RemoteWebDriver fromReusedSession(URL remoteAddress, Capabilities
     protected ReusableRemoteWebDriver(CommandExecutor executor, Capabilities capabilities, SessionId sessionId) {
         super();
         setCommandExecutor(executor);
+        setReusedCapabilities(capabilities);
         setSessionId(sessionId.toString());
+    }
+
+    protected ReusableRemoteWebDriver(URL remoteAddress, Capabilities capabilities, SessionId sessionId) {
+        super();
+        HttpCommandExecutor httpCommandExecutor = new HttpCommandExecutor(remoteAddress);
+        setCommandExecutor(httpCommandExecutor);
         setReusedCapabilities(capabilities);
+
+        setValueToFieldInHttpCommandExecutor(httpCommandExecutor, "commandCodec", Dialect.OSS.getCommandCodec());
+        setValueToFieldInHttpCommandExecutor(httpCommandExecutor, "responseCodec", Dialect.OSS.getResponseCodec());
```

Comment:
Because of this commit: https://github.com/SeleniumHQ/selenium/commit/d725150ac7a9ca5c250b075d66cf7a3caeb32413
The executors are required not to be null when some command (other than the one for creating a new session) is executed: https://github.com/SeleniumHQ/selenium/blob/master/java/client/src/org/openqa/selenium/remote/HttpCommandExecutor.java#L152
but these executors are created only when a new session is being created - which is not the case when we are using "reused session". I tried to create a new session and then to replace the sessionId with the older one, but then there was hanging the additional sessions which caused failures.